internal/broker/broker_test.go

```diff
@@ -615,39 +622,38 @@ func TestIsAuthenticated(t *testing.T) {
 // Due to ordering restrictions, this test can not be run in parallel, otherwise the routines would not be ordered as expected.
 func TestConcurrentIsAuthenticated(t *testing.T) {
 	tests := map[string]struct {
-		firstUser  string
-		secondUser string
+		firstCallDelay  int
+		secondCallDelay int
 
 		timeBetween time.Duration
 	}{
-		"First auth starts and finishes before second":                  {timeBetween: 2 * time.Second},
-		"First auth starts first but second finishes first":             {firstUser: "user-timeout-3", timeBetween: time.Second},
-		"First auth starts first then second starts and first finishes": {firstUser: "user-timeout-2", secondUser: "user-timeout-3", timeBetween: time.Second},
+		"First auth starts and finishes before second":                  {secondCallDelay: 1, timeBetween: 2 * time.Second},
+		"First auth starts first but second finishes first":             {firstCallDelay: 3, timeBetween: time.Second},
+		"First auth starts first then second starts and first finishes": {firstCallDelay: 2, secondCallDelay: 3, timeBetween: time.Second},
```

Comment:
That's now finally done. I rebased the PR on https://github.com/ubuntu/authd-oidc-brokers/pull/171 and added another commit "refactor: Use fewer magic string constants in tests" to achieve that without making the tests more complicated.

Because i had to rebase anyway, I also squashed the fixup commits and resolved a lot of conflicts, both with main and https://github.com/ubuntu/authd-oidc-brokers/pull/171. I hope I didn't mess up any merge conflict resolution, but I gave my best to be thorough (e.g. `golangci-lint` and `go test ./internal/broker/...` pass on all commits).